Ubuntu: https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/386922
Upstream: not yet, but probably can and should be
Description: For non-recovery kernel menu entries, suppress progress
 messages as the kernel and initrd load.

Index: b/util/grub.d/10_linux.in
===================================================================
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -82,13 +82,21 @@
     prepare_boot_cache="$(prepare_grub_to_access_device ${GRUB_DEVICE_BOOT} | sed -e "s/^/\t/")"
   fi
   printf '%s\n' "${prepare_boot_cache}"
-  cat << EOF
+  if [ "x$5" != "xquiet" ]; then
+    cat << EOF
 	echo	'$(printf "$(gettext_quoted "Loading Linux %s ...")" ${version})'
+EOF
+  fi
+  cat << EOF
 	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
 EOF
-  if test -n "${initrd}" ; then
+  if [ "x$5" != "xquiet" ]; then
     cat << EOF
 	echo	'$(gettext_quoted "Loading initial ramdisk ...")'
+EOF
+  fi
+  if test -n "${initrd}" ; then
+    cat << EOF
 	initrd	${rel_dirname}/${initrd}
 EOF
   fi
@@ -130,7 +138,8 @@
   fi
 
   linux_entry "${OS}" "${version}" false \
-      "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_EXTRA} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+      "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_EXTRA} ${GRUB_CMDLINE_LINUX_DEFAULT}" \
+      quiet
   if [ "x${GRUB_DISABLE_LINUX_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" true \
 	"single ${GRUB_CMDLINE_LINUX}"
